File: Garbagesorting/app/src/main/java/dk/itu/garbagesorting/ItemsDB.java
package dk.itu.garbagesorting;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class ItemsDB {
    private static ItemsDB sItemsDB;
    private Map<String, String> itemsDB= new HashMap<>();

    private ItemsDB() { fillItemsDB(); }

    public static void initialize() {
        if (sItemsDB == null) sItemsDB= new ItemsDB();
    }

    public static ItemsDB get() {
        if (sItemsDB == null) throw new IllegalStateException("ItemsDB must be initialized");
        return sItemsDB;
    }

    public void addItem(String what, String where){
        itemsDB.put(what, where);
    }

    public void fillItemsDB() {
        itemsDB.put("coffee", "residual waste");
        itemsDB.put("carrots", "residual waste");
        itemsDB.put("milk carton", "cardboard");
        itemsDB.put("bread", "residual waste");
        itemsDB.put("butter", "residual waste");
        itemsDB.put("meat package", "plastic");
        itemsDB.put("bean can", "metal");
    }

    public String searchForItem(String item){
        String itemPlace = "not found";

        for (Map.Entry<String, String> existingItem : itemsDB.entrySet()) {
            if (existingItem.getKey().equals(item)){
                itemPlace = existingItem.getValue();
            }
        }
        return itemPlace;
    }

    public Map<String, String> getMap(){
        return itemsDB;
    }
}

File: Garbagesorting/app/src/main/java/dk/itu/garbagesorting/ItemsDB.kt
package dk.itu.garbagesorting

import android.content.Context
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader

class ItemsDB private constructor(context: Context) {
    private val itemsDB: MutableMap<String, String> = HashMap()

    init {
        fillItemsDB(context)
    }

    fun addItem(what: String, where: String) {
        itemsDB[what] = where
    }

    fun fillItemsDB(context: Context) {
        try {
            val reader = BufferedReader(
                    InputStreamReader(context.assets.open("garbage.txt")))
            var line = reader.readLine()
            while (line != null) {
                val listItem = line.split(", ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
                addItem(listItem[0], listItem[1])
                line = reader.readLine()
            }
        } catch (e: IOException) {
            println("Error")
        }
    }

    fun searchForItem(item: String): String {
        var itemPlace = "not found"
        for ((key, value) in itemsDB) {
            if (key == item) {
                itemPlace = value
            }
        }
        return itemPlace
    }

    val map: Map<String, String>
        get() = itemsDB

    companion object {
        private var sItemsDB: ItemsDB? = null
        fun initialize(context: Context) {
            if (sItemsDB == null) sItemsDB = ItemsDB(context)
        }

        fun get(): ItemsDB? {
            checkNotNull(sItemsDB) { "ItemsDB must be initialized" }
            return sItemsDB
        }
    }
}